### PR TITLE
Update ReactS3Uploader.js

### PR DESCRIPTION
**Pull request by alexprice91**
Remove potential non-native props from the input.

This is for thew new react warning:
https://facebook.github.io/react/warnings/unknown-prop.html

### CODE REVIEW

File: ReactS3Uploader.js [hunk 1]
```diff
@@ -55,8 +55,23 @@ var ReactS3Uploader = React.createClass({
         clearInputFile(ReactDOM.findDOMNode(this));
     },
 
+    getNewProps() {
```
Comment (seanadkinson) [inline ReactS3Uploader.js]: Let's name this `getInputProps()`, since that's what it is doing.

File: ReactS3Uploader.js [hunk 2]
```diff
@@ -55,8 +55,23 @@ var ReactS3Uploader = React.createClass({
...
+        var temporaryProps = objectAssign({}, this.props, {type: 'file', onChange: this.uploadFile});
```
Comment (seanadkinson) [inline ReactS3Uploader.js]: Name this `allProps`

File: ReactS3Uploader.js [hunk 3]
```diff
@@ -55,8 +55,23 @@ var ReactS3Uploader = React.createClass({
...
+        var newProps = {};
```
Comment (seanadkinson) [inline ReactS3Uploader.js]: And this `inputProps`

File: ReactS3Uploader.js [hunk 4]
```diff
@@ -55,8 +55,23 @@ var ReactS3Uploader = React.createClass({
...
+                newProps[key] = temporaryProps[key];
+            }
+        }
+
+        return newProps;
+    },
+
     render: function() {
-        return React.DOM.input(objectAssign({}, this.props, {type: 'file', onChange: this.uploadFile}));
+        return React.DOM.input(this.getNewProps());
```
Comment (seanadkinson) [inline ReactS3Uploader.js]: Can you move the method below `render`?  Usually you like to see methods be called top-down in the file - https://grysz.com/2015/11/13/reformat-java-code-in-intellij-according-to-uncle-bobs-stepdown-rule/

**Comment by seanadkinson** [comment]
Bunch of nit-picking, but thanks! 👍 

Will merge if you can make these few changes.

**Comment by alexprice91** [comment]
@seanadkinson can never be too thorough, especially in our profession. I implemented it.

**Comment by seanadkinson** [comment]
Thanks 👍

**Pull request merged**